### PR TITLE
implement missing video methods, *Must* be merged before flash removal

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description
* Implement missing methods on Player, HTML, and Flash
* Remove methods from Flash that are not needed
* Added unit tests for all API only methods

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [x] Unit Tests updated or fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/js/player.js [hunk 1]
```diff
@@ -1984,6 +1996,40 @@ class Player extends Component {
...
+   *
+   * @return {boolean|Player}
+   *         - true if defaultMuted is on and getting
+   *         - false if defaultMuted is off and getting
+   *         - A reference to the current player when setting
+   */
+  defaultMuted(defaultMuted) {
+    if (defaultMuted !== undefined) {
+      this.techCall_('setDefaultMuted', defaultMuted);
+      return this;
```
Comment (misteroneill) [inline src/js/player.js]: Assuming we continue with the plan to remove chaining support, we may want to change this.

**Comment by gkatsev** [comment]
https://github.com/BrandonOCasey/video.js/pull/5/files

**Comment by gkatsev** [comment]
I think this potentially could be merged as a minor?

File: src/js/player.js [hunk 2]
```diff
@@ -3235,13 +3306,6 @@ class Player extends Component {
     return this.tech_ && this.tech_.videoHeight && this.tech_.videoHeight() || 0;
   }
 
-  // Methods to add support for
-  // initialTime: function() { return this.techCall_('initialTime'); },
```
Comment (gkatsev) [inline src/js/player.js]: are `initialTime` and `startOffsetTime` no longer things?

File: src/js/tech/html5.js [hunk 3]
```diff
@@ -1193,6 +1193,21 @@ Html5.resetMediaElement = function(el) {
...
+   * will indicate the current status of the volume and `defaultMuted` will not.
+   *
+   * @method Html5.prototype.defaultMuted
+   * @return {boolean}
+   *         - True if the value of `volume` should be ignored and the audio set to silent.
+   *         - False if the value of `volume` should be used.
+   *
+   * @see [Spec]{@link https://www.w3.org/TR/html5/embedded-content-0.html#dom-media-defaultmuted}
+   */
+  'defaultMuted',
```
Comment (gkatsev) [inline src/js/tech/html5.js]: these getters/setters need corresponding empty methods in `tech.js`

File: src/js/player.js (hunk 2 above)
Comment (misteroneill) [inline src/js/player.js]: Neither appear to be in the standard.

File: src/js/tech/html5.js (hunk 3 above)
Comment (misteroneill) [inline src/js/tech/html5.js]: Good catch.

Comment (BrandonOCasey) [inline src/js/tech/html5.js]: we don't have empty getters/setters for any of the old methods. Do we want to add them across the board?

**Comment by gkatsev** [comment]
Fixes https://github.com/videojs/video.js/issues/523

File: src/js/tech/html5.js (hunk 3 above)
Comment (gkatsev) [inline src/js/tech/html5.js]: I guess it's fine as long as it doesn't cause unnecessary errors (and we can always add them in later).

**Comment by mrocajr** [comment]
LGTM

**Comment by kavin-90** [comment]
I didn't saw pause show big play button after using stackoverflow code it works.

**Pull request merged**